### PR TITLE
CB-12485 crn region needs to be configurable

### PR DESCRIPTION
**Pull request by horadla23**
See detailed description in the commit message.

### CODE REVIEW

**Comment by horadla23** [comment]
(sidenote: it seems code owner feature for authorization module is working :) )

File: common/src/main/java/com/sequenceiq/cloudbreak/auth/crn/CrnGeneratorService.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.auth.altus.Crn;
+import com.sequenceiq.cloudbreak.auth.altus.CrnResourceDescriptor;
+
+@Service
+@ConfigurationProperties("crn")
+public class CrnGeneratorService {
```
Comment (lnardai) [inline common/src/main/java/com/sequenceiq/cloudbreak/auth/crn/CrnGeneratorService.java:13]: How about RegionAwareCrnGenerator? I would like to convey what is the usecase for this service vs. the regular Crn.Builder.

Comment (horadla23) [reply common/src/main/java/com/sequenceiq/cloudbreak/auth/crn/CrnGeneratorService.java:13]: I like it, I will change the class name

**Review by lnardai — APPROVED** [review]
Looks good! 
Only question in my mind is how do we enforce people to use the 'RegionAwareCrn' builder. It would be nice to restrict, at least in application code that only the proper CRN with correct region should be used. 
We should at least make it hard for people to create Crn's with the default region and partition. Any ideas on that?

**Comment by horadla23** [comment]
retest this please

**Pull request merged**